interface/drugs/add_edit_drug.php

```diff
@@ -176,55 +182,50 @@ function sel_related() {
     $new_drug = false;
     if ($drug_id) {
         if ($_POST['form_save']) { // updating an existing drug
-            sqlStatement("UPDATE drugs SET " .
-            "name = '"           . escapedff('form_name')          . "', " .
-            "ndc_number = '"     . escapedff('form_ndc_number')    . "', " .
-            "drug_code = '"      . escapedff('form_drug_code')    . "', " .
-            "on_order = '"       . escapedff('form_on_order')      . "', " .
-            "reorder_point = '"  . escapedff('form_reorder_point') . "', " .
-            "max_level = '"      . escapedff('form_max_level')     . "', " .
-            "form = '"           . escapedff('form_form')          . "', " .
-            "size = '"           . escapedff('form_size')          . "', " .
-            "unit = '"           . escapedff('form_unit')          . "', " .
-            "route = '"          . escapedff('form_route')         . "', " .
-            "cyp_factor = '"     . numericff('form_cyp_factor')    . "', " .
-            "related_code = '"   . escapedff('form_related_code')  . "', " .
-            "allow_multiple = "  . (empty($_POST['form_allow_multiple' ]) ? 0 : 1) . ", " .
-            "allow_combining = " . (empty($_POST['form_allow_combining']) ? 0 : 1) . ", " .
-            "active = "          . (empty($_POST['form_active']) ? 0 : 1) . " " .
-            "WHERE drug_id = ?", array($drug_id));
-            sqlStatement("DELETE FROM drug_templates WHERE drug_id = ?", array($drug_id));
+           $updateDrug = array(
+            "name" => escapedff('form_name'),
+            "ndc_number" => escapedff('form_ndc_number'),
+            "drug_code" => escapedff('form_drug_code'),
+            "on_order" => escapedff('form_on_order'),
+            "reorder_point" => escapedff('form_reorder_point'),
+            "max_level" => escapedff('form_max_level'),
+            "form" => escapedff('form_form'),
+            "size" => escapedff('form_sixe'),
+            "unit" => escapedff('form_unit'),
+            "route" => escapedff('form_route'),
+            "cyp_factor" => numericff('form_cyp_factor'),
+            "related_code" => escapedff('form_related_code'),
+            "allow_multiple" => (empty($_POST['form_allow_multiple' ]) ? 0 : 1),
+            "allow_combining" => (empty($_POST['form_allow_combining']) ? 0 : 1),
+            "active" => (empty($_POST['form_active']) ? 0 : 1)      
```

Comment:
remove all the escapedff (and change the numericff to a trim()) and instead use sql binding in the function in the service class.